src/main/java/io/cdap/plugin/jms/source/JMSSourceUtils.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © 2016-2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.jms.source;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.etl.api.streaming.StreamingContext;
+import io.cdap.plugin.jms.common.JMSConfig;
+import io.cdap.plugin.jms.common.JMSMessageType;
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.api.java.JavaDStream;
+import org.apache.spark.streaming.receiver.Receiver;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import javax.jms.BytesMessage;
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+import javax.jms.Message;
+import javax.jms.ObjectMessage;
+import javax.jms.TextMessage;
+
+/**
+ * An utils class that encapsulates the necessary functionality to convert different types of JMS messages to
+ * StructuredRecords, and creates an input stream with our customized implementation of receiver {@link JMSReceiver}.
+ */
+public class JMSSourceUtils {
+
+  static JavaDStream<StructuredRecord> getJavaDStream(StreamingContext context,
+                                                      JMSConfig config) {
+    Receiver<StructuredRecord> jmsReceiver = new JMSReceiver(StorageLevel.MEMORY_AND_DISK_SER_2(), config);
+    return context.getSparkStreamingContext().receiverStream(jmsReceiver);
+  }
+
+  public static StructuredRecord convertMessage(Message message, JMSConfig config) throws JMSException,
+    IllegalArgumentException {
+    if (message instanceof BytesMessage && config.getMessageType().equals(JMSMessageType.BYTES.getName())) {
+      return convertByteMessage(message, config);
+    } else if (message instanceof MapMessage && config.getMessageType().equals(JMSMessageType.MAP.getName())) {
```

Comment:
No need to have `else` after an `if` block that returns. It is cleaner to have:

```
if (...) {
  return ...;
}
if (...) {
  return ...;
}
```

Reply:
Agree. Fixed this.